docs/code-style.md

```diff
@@ -0,0 +1,333 @@
+child component be slightly different and break uniformed UI/UX.
+
+Some most basic components might still use `classNames` to inject classes and be
+in sync with React's own naming. But this should be considered a rare exception
+that deals with most generic components and as a workaround to a greater
+problem. Example:
+
+```tsx
+<ButtonNG
+  className="AdminRestricted__switch-button"
+  variant="primary"
+  loading={isLoggingOut}
+  disabled={isLoggingOut}
+  onClick={handleLogout}
+>
+```
+
+The way to "inject" some classes but still keep the number of variations to the
+minimum is a variance prop. Such example is the above `variant="primary"`, but
+not limited.
+
+Example: A component can have prop named
+
+```tsx
+width ? : "wide" | "narrow" | "normal"
+```
+
+and those strings be the names of exported constants in `Component.tailwind.ts`.
+These classes can be accessed as `TW[width]` in a code like:
+
+```tsx
+const componentClasses = cn(className, TW.common, {
+  [TW[variant]]: variant,
+  [TW[width]]: width,
+  [TW.disabled]: disabled,
+  [TW.enabled]: !disabled,
+  [TW.loading]: loading,
+  [TW.notLoading]: !loading,
+});
+```
+
+## TypeScript code (coding)
+
+## Imports
+
+Prefer shorthand (absolute) imports over relative ones. You can access every
+directory under `. /src` with just typing its name. Example: instead of
+`../../../../settings`, you can write`settings`.
+
+##
+
+Exports
+
+Should try to not export types and interfaces that aren 't used elsewhere
+outside that single file.There is a tool(`npm run prune`) that can generate a
+list of all unused exports.This tool might be used alongside the linters that
+execute before each commit.As for components, if they're only imported by a
+sibling `index.ts` file, then there should be a check if that component is used
+anywhere and if it's good to keep or delete it.
+
+###
+
+Types vs Interfaces
+
+Usually types declared with `interface` were used to define a components
+props.However`type` s syntax allows for more robust definitions and should be
+preferred in most cases and will allow for more uniformed looking code.
+
+###
+
+Enums
+
+A simplistic explanation of TypeScript is it removes type information upon
+compilation.The formal words used in the documentation is "type-level extension"
+since most TS additions are type - level extensions to JS, and they don 't
+affect the code' s runtime behavior.TypeScript 's solution in the enum case is
+to break its own rule. When compiling it, extra JS code is emitted that never
+existed in the original TS code.This adds a confusing complication to the
+otherwise simple compiler model.For this reason, recommendation is to avoid
+enums in favor of using unions.Example :
+
+```ts
+enum AllowedColor {
+  red = "red",
+  green = "green",
+  blue = "blue",
+}
+```
+
+would be
+
+```ts
+type AllowedColor = "red" | "green" | "blue";
+```
+
+For extensive range in values, branded types could be more useful.
```

Comment:
Disagree with this. @mike-lvov does too if I remember right.
Enums are a feature of the language we are using. We shouldn't be afraid to use those features.

Reply:
**TL;DR**:

If there is a need to remove the passage from this document, be it for it causing worse code, adding more confusion, stating in-factual claims or simply just taking up space, we can do so with a next PR.

----

The longer part (recorded here so it will be safe to forget):

My idea isn't to remove passages though, but add more, maybe even to the point of a single document like this one becoming a reference for several ones covering more specific topics. 

In that light, I was more interested if I should put some naming suggestions for the `enums` we're already using and might be using in the future since right now their casings are all over the place. I just figured I might be overdoing it to try to account for every single naming in this first draft.

On the comment above, I don't know what precisely `this` is being disagreed to, I'm guessing `enums` usage,
but if there is a signal of an issue with this document, it can resolved by means mentioned above.

However, the comment has several notions I find puzzling and can answer only by a bit of deconstruction:
- The `Enums` passage is a simplified explanation of how the language works, meant to be factual as to explain the recommendation;
- Just like the entire document isn't a mandate, but a collection of suggestions, ideas and examples as opposed to restrictions, thus if anything should be followed, I'd like it to be the very first one: `use your own best judgement`;
- I define feature as only what one notices and not every functionality is a feature, but only those different persons notice or are significant to them. Writing this point because of the examples that follow;
- Not every functionality of a language must be used, like:
  - `namespace` is one of TypeScript, yet not a single use in the codebase - we just don't notice it;
  -  `with` is one of JavaScript, we all stay clear of it - our lives aren't worse for it;
- I find "don't be afraid to use it" a misleading argument since I don't see a reason to input emotions like "fear" into a place there aren't.
 
We can simply use `enum` or not, and as long as we know the pros and cons of it, I think it's good enough.